cmd/kubeadm/app/phases/certs/renewal/readwriter.go

```diff
@@ -141,11 +140,15 @@ func (rw *kubeConfigReadWriter) Read() (*x509.Certificate, error) {
 	// For local CA renewal, the local CA on disk could have changed, thus a reload is needed.
 	// For CSR renewal we assume the same CA on disk is mounted for usage with KCM's
 	// '--cluster-signing-cert-file' flag.
-	caCert, _, err := certsphase.LoadCertificateAuthority(rw.certificateDir, rw.baseName)
+	certificatePath, _ := pkiutil.PathsForCertAndKey(rw.certificateDir, rw.baseName)
+	caCerts, err := certutil.CertsFromFile(certificatePath)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to load existing certificate %s", rw.baseName)
+	}
+	if len(caCerts) != 1 {
+		return nil, errors.Errorf("wanted exactly one certificate, got %d", len(caCerts))
 	}
-	rw.caCert = caCert
+	rw.caCert = caCerts[0]
```

Comment:
note, this is already an assumption in a number of places - e.g
https://github.com/kubernetes/kubernetes/blob/847b2e1085f9f642292da9f3f1a56ffb5366ab86/cmd/kubeadm/app/phases/certs/renewal/readwriter.go#L89